core/src/main/resources/reference.conf

```diff
@@ -341,6 +341,20 @@ datastax-java-driver {
 
   # ADVANCED OPTIONS -------------------------------------------------------------------------------
 
+  # The maximum number of live sessions that are allowed to coexist in a given VM.
+  #
+  # This is intended to help detect resource leaks in client applications that create too many
+  # sessions and/or do not close them correctly. The driver keeps track of the number of live
+  # sessions in a static variable; if it gets over this threshold, a warning will be logged for
+  # every new session.
+  #
+  # If the value is less than or equal to 0, the feature is disabled: no warning will be issued.
+  #
+  # Required: yes
+  # Modifiable at runtime: yes, the new value will be used for sessions created after the change.
+  # Overridable in a profile: no
+  advanced.session-leak.threshold = 4
```

Comment:
I think the default should be aggressive. Most clients probably use 1 anyway, so it's better to warn fast.